packages/sourcecred/src/core/credGrainView.js

```diff
@@ -131,20 +131,70 @@ export class CredGrainView {
   }
 
   validateForGrainAllocation() {
+    this.activeParticipants().map((p) => {
+      p.credPerInterval.map((c) => {
+        if (typeof c !== "number") {
+          throw new Error(`Non numeric cred value found`);
+        }
+      });
+
+      if (typeof p.cred !== "number") {
+        throw new Error(`Non numeric cred value found`);
+      }
+    });
+
+    this.activeParticipants().map((p) => {
+      if (p.credPerInterval.length !== this.intervals().length) {
+        throw new Error(`participant cred per interval length mismatch`);
+      }
+    });
+
+    this.activeParticipants().map((p) => {
+      if (p.grainEarnedPerInterval.length !== this.intervals().length) {
+        throw new Error(`participant grain per interval length mismatch`);
+      }
+    });
```

Comment:
It'd be more efficient and concise if you placed all checks into the same loops. Not sure why the same loops are being duplicated throughout this function.

Reply:
fixed